builtin/providers/google/resource_compute_instance_template.go

```diff
@@ -305,11 +305,9 @@ func buildNetworks(d *schema.ResourceData, meta interface{}) (error, []*compute.
 	for i := 0; i < networksCount; i++ {
 		prefix := fmt.Sprintf("network_interface.%d", i)
 
-		source := "global/networks/default"
+		source := "global/networks/"
```

Comment:
I think the right thing to do here is similar to what I did with the target pool: https://github.com/hashicorp/terraform/blob/master/builtin/providers/google/resource_compute_target_pool.go#L118

I.e. if it's a simple name, build the URL client side (don't look up the network as it might not actually exist yet).  But let you use the URL as well.